idea_town/experiments/models.py

```diff
@@ -0,0 +1,52 @@
+from django.db import models
+from django.contrib.auth.models import User
+
+from ..utils import HashedUploadTo
+
+
+experiment_thumbnail_upload_to = HashedUploadTo('thumbnail')
+experimentdetail_image_upload_to = HashedUploadTo('image')
+
+
+class Experiment(models.Model):
+
+    title = models.CharField(max_length=128)
+    slug = models.SlugField(max_length=128, unique=True, db_index=True)
+    thumbnail = models.ImageField(upload_to=experiment_thumbnail_upload_to)
+    description = models.TextField()
+
+    users = models.ManyToManyField(User, through='UserInstallation')
+
+    created = models.DateTimeField(auto_now_add=True)
+    modified = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return self.title
+
+
+class ExperimentDetail(models.Model):
```

Comment:
Well, we'll definitely need the experiments to be signed by AMO. AMO has an auto-signing tool built into the submission process; it was really easy to get the universal-search addon to pass the auto-review, and you get a private AMO URL for free along with the signing. That seems like the simplest starting point for hosting addons?